### PR TITLE
feat: Openrag improved tasks logs backend

### PR DESCRIPTION
**Pull request by ricofurtado**
This pull request introduces a more robust and user-friendly workflow for handling duplicate file uploads, especially when syncing files to connectors or uploading folders. The main improvements include duplicate detection before syncing, a dialog that lists duplicate filenames, and options for users to overwrite or skip duplicates. Additionally, several new CSS variables and a new icon component were added to support these UI enhancements.

**Duplicate Handling Workflow Improvements:**

- Added pre-upload duplicate checks for selected files when syncing to connectors, prompting the user with a dialog listing duplicate filenames and options to overwrite or skip them. This prevents accidental overwrites and improves user awareness. ([frontend/app/upload/[provider]/page.tsxR415-R423](diffhunk://#diff-f01b7cca092945fb6177f8ce8c2cb76f4f15f626647184f617e87a8b497548ceR415-R423), [frontend/app/upload/[provider]/page.tsxL430-R451](diffhunk://#diff-f01b7cca092945fb6177f8ce8c2cb76f4f15f626647184f617e87a8b497548ceL430-R451), [frontend/app/upload/[provider]/page.tsxR460](diffhunk://#diff-f01b7cca092945fb6177f8ce8c2cb76f4f15f626647184f617e87a8b497548ceR460), [frontend/app/upload/[provider]/page.tsxR478-R560](diffhunk://#diff-f01b7cca092945fb6177f8ce8c2cb76f4f15f626647184f617e87a8b497548ceR478-R560), [frontend/app/upload/[provider]/page.tsxL655-R749](diffhunk://#diff-f01b7cca092945fb6177f8ce8c2cb76f4f15f626647184f617e87a8b497548ceL655-R749), [frontend/app/upload/[provider]/page.tsxR769-R776](diffhunk://#diff-f01b7cca092945fb6177f8ce8c2cb76f4f15f626647184f617e87a8b497548ceR769-R776), [[1]](diffhunk://#diff-d146a917c64de751accfaf7bac521e021949a3b44b97c29076aa93e6ef0ee0c9R22-R26) [[2]](diffhunk://#diff-d146a917c64de751accfaf7bac521e021949a3b44b97c29076aa93e6ef0ee0c9R36-R69) [[3]](diffhunk://#diff-d146a917c64de751accfaf7bac521e021949a3b44b97c29076aa93e6ef0ee0c9R80-R92)

- Enhanced the `DuplicateHandlingDialog` to display up to five duplicate filenames and indicate if there are more, making it easier for users to review which files are affected. [[1]](diffhunk://#diff-d146a917c64de751accfaf7bac521e021949a3b44b97c29076aa93e6ef0ee0c9R22-R26) [[2]](diffhunk://#diff-d146a917c64de751accfaf7bac521e021949a3b44b97c29076aa93e6ef0ee0c9R36-R69) [[3]](diffhunk://#diff-d146a917c64de751accfaf7bac521e021949a3b44b97c29076aa93e6ef0ee0c9R80-R92)

- Updated the folder upload duplicate handling logic in `KnowledgeDropdown` to track and display duplicate filenames instead of just counts, improving feedback and consistency with connector uploads. [[1]](diffhunk://#diff-62ce844c2ab79fff41954cc3d4b66de45ef3af6c7c6b4b237c0f5a73d9d28d40L145-R145) [[2]](diffhunk://#diff-62ce844c2ab79fff41954cc3d4b66de45ef3af6c7c6b4b237c0f5a73d9d28d40L385-R391) [[3]](diffhunk://#diff-62ce844c2ab79fff41954cc3d4b66de45ef3af6c7c6b4b237c0f5a73d9d28d40L430-R436) [[4]](diffhunk://#diff-62ce844c2ab79fff41954cc3d4b66de45ef3af6c7c6b4b237c0f5a73d9d28d40L518-R521) [[5]](diffhunk://#diff-62ce844c2ab79fff41954cc3d4b66de45ef3af6c7c6b4b237c0f5a73d9d28d40L539-R540) [[6]](diffhunk://#diff-62ce844c2ab79fff41954cc3d4b66de45ef3af6c7c6b4b237c0f5a73d9d28d40L859-R860)

**API and Mutation Updates:**

- Extended the `useSyncConnector` mutation to accept a `replace_duplicates` flag, allowing the backend to distinguish between overwrite and skip actions for duplicates. ([frontend/app/api/mutations/useSyncConnector.tsR75-R76](diffhunk://#diff-2a23e600b729ba22fdb6a9bc34add47856d755474a1604b8342974de8cbbd4fdR75-R76), [frontend/app/upload/[provider]/page.tsxR460](diffhunk://#diff-f01b7cca092945fb6177f8ce8c2cb76f4f15f626647184f617e87a8b497548ceR460))

**UI and Design Enhancements:**

- Added new CSS variables for task status colors and duplicate/failure states to improve visual feedback in the UI. [[1]](diffhunk://#diff-6de52daad7039d252d605df9a4b8a8c44af6ab6750b6284879b638c3d5d1c632R66) [[2]](diffhunk://#diff-6de52daad7039d252d605df9a4b8a8c44af6ab6750b6284879b638c3d5d1c632R92) [[3]](diffhunk://#diff-6de52daad7039d252d605df9a4b8a8c44af6ab6750b6284879b638c3d5d1c632R161) [[4]](diffhunk://#diff-6de52daad7039d252d605df9a4b8a8c44af6ab6750b6284879b638c3d5d1c632R244-R250) [[5]](diffhunk://#diff-6de52daad7039d252d605df9a4b8a8c44af6ab6750b6284879b638c3d5d1c632R342-R348)

- Added a new `IncidentReporterIcon` React component for consistent iconography related to incident reporting.

**Other Minor Updates:**

- Changed the OAuth prompt parameter from `select_account` to `consent` in the connector authorization flow to better align with user expectations.

These changes collectively make file uploads and syncs safer and more transparent for users, while also improving the look and feel of the app.

## Summary by CodeRabbit

* **New Features**
  * New "enhanced" task endpoints that return per-file structured failure metadata to aid diagnosis.

* **Bug Fixes**
  * Polling now retries transient result-fetch errors and surfaces clearer failed outcomes for permanent errors and timeouts.
  * Routing updated so literal "enhanced" paths are not misinterpreted as task IDs.

* **Tests**
  * Added and updated tests for failure metadata, retry semantics, and polling edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/langflow-ai/openrag/pull/1671?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: frontend/components/task-error-content.tsx [hunk 1]
```diff
@@ -27,126 +36,199 @@ export function TaskErrorContent({
...
+        className="inline-flex shrink-0 items-center justify-center text-muted-foreground hover:text-foreground"
+        onClick={(event) => {
+          event.preventDefault();
+          event.stopPropagation();
+        }}
+        onPointerDown={(event) => event.stopPropagation()}
+      >
+        <IncidentReporterIcon className="size-4" />
+      </button>
+    </div>
```
[the remark this answers is not in the document]
Comment (Wallgau) [reply frontend/components/task-error-content.tsx:93]: I'll take care of those ones in next PR, not current change here

File: frontend/app/globals.css [hunk 2]
```diff
@@ -156,6 +158,7 @@
     --failure-message: #ffc9c9cc;
     --failure-scroll: #ff6464;
     --failure-muted: #a66262;
+    --failure-component-cause: 0 0% 32%; /* #525252 */
```
[the remark this answers is not in the document]
Comment (Wallgau) [reply frontend/app/globals.css:161]: I'll take care of those ones in next PR, not current change here

[the remark this answers is not in the document]
Comment (Wallgau) [reply frontend/app/globals.css:161]: No

File: frontend/components/task-error-content.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Wallgau) [reply frontend/components/task-error-content.tsx:93]: No

File: src/api/v1/documents.py [hunk 3]
```diff
@@ -4,23 +4,24 @@
 Provides document ingestion and management.
 Uses API key authentication.
 """
+
 from typing import List, Optional
```
[the remark this answers is not in the document]
Comment (mpawlow) [reply src/api/v1/documents.py]: ### (1a) [Major] Unused `List` and `Optional` typing imports block CI Ruff checks

- Concur with **CodeRabbit**

#### Problem

- `src/api/v1/documents.py` line 8 retains `from typing import List, Optional` even though the `ingest_endpoint` signature was updated in the same PR to use modern union-type syntax (`list[UploadFile]`, `str | None`).
- Neither `List` nor `Optional` is referenced anywhere in the file.
- This triggers two separate Ruff failures (`F401` unused import, `UP035` deprecated alias) and blocks the `Lint Backend / Ruff and mypy on changed files` CI check.

#### Potential Solution

Remove the entire import line:

```diff
-from typing import List, Optional
```

File: src/services/task_service.py [hunk 4]
```diff
@@ -598,21 +726,13 @@ def get_task_status(self, user_id: str, task_id: str) -> dict | None:
...
-                "created_at": file_task.created_at,
-                "updated_at": file_task.updated_at,
-                "duration_seconds": file_task.duration_seconds,
-                "filename": file_task.filename,
-                "phase": file_task.phase.value,
-                "docling_status": file_task.docling_status.value,
-                "docling_task_id": file_task.docling_task_id,
-            }
+            entry = self._serialize_file_task(file_task)
+            if file_task.status == TaskStatus.FAILED:
```
Comment (mpawlow) [inline src/services/task_service.py:730]: ### (1b) [Normal] Mixed status comparison styles in new service methods

#### Problem

- The two new methods — `get_task_status2` and `get_all_tasks2` — inconsistently check task/file status:
  - `file_task.status == TaskStatus.FAILED` (enum identity comparison)
  - `file_task.status.value == "running"` (string literal comparison)
  - `file_task.status.value == "pending"` (string literal comparison)
  - `file_task.status.value != "completed"` (string literal comparison)
- Within the same code block, enum comparisons and `.value` string comparisons are mixed.
- String comparisons couple logic to the string representation of the enum and would silently break if an enum member value ever changes.

#### Code References

- `src/services/task_service.py:736–738` — `if file_task.status.value == "running":` / `elif file_task.status.value == "pending":`
- `src/services/task_service.py:777` — `if file_task.status.value != "completed":`
- `src/services/task_service.py:785–787` — same string patterns in `get_all_tasks2`

#### Potential Solution

Standardise on enum comparisons throughout both methods:

```python
# instead of:
if file_task.status.value == "running":
    running_files_count += 1
elif file_task.status.value == "pending":
    pending_files_count += 1

# use:
if file_task.status == TaskStatus.RUNNING:
    running_files_count += 1
elif file_task.status == TaskStatus.PENDING:
    pending_files_count += 1
```

And replace `file_task.status.value != "completed"` with `file_task.status != TaskStatus.COMPLETED`.

File: src/services/task_service.py [hunk 5]
```diff
@@ -569,26 +570,153 @@ async def process_with_semaphore(item, item_key: str):
...
+
+        if phase == IngestionPhase.DOCLING and docling_status == DoclingPhaseStatus.PROCESSING:
+            return {
+                "component": "docling",
+                "failure_phase": "parsing",
+                "user_facing_message": "Document processing timed out. Please retry ingestion.",
+                "actionable_by": "RETRYABLE",
+            }
+
+        if phase == IngestionPhase.LANGFLOW:
```
Comment (mpawlow) [inline src/services/task_service.py:660]: ### (1c) [Normal] `_infer_failure_metadata` ordering: LANGFLOW phase check shadows the `"already exists"` branch

#### Problem

- The priority-ordered early-return chain in `_infer_failure_metadata` places the `phase == IngestionPhase.LANGFLOW` check (item 5) **before** the `"already exists" in error` check (item 6).
- A file whose failure occurs during the Langflow phase but whose `error` string contains `"already exists"` will be classified as `langflow / unknown / RETRYABLE` instead of `openrag / file_validation / USER_ACTIONABLE`.
- The ordering implies "if it failed during Langflow, the component is always langflow/unknown" — but duplicate-file rejections propagated via the Langflow error message would silently be misclassified.
- No test covers this ordering edge case.

#### Background Information

- The docstring says "Priority order: docling_status enum first (stable), error string patterns second (fallback for edge cases)." The LANGFLOW branch should arguably rank below the error-pattern checks if those patterns represent globally valid classifications regardless of phase.

#### Potential Solution

Move the `"already exists"` (and any other globally-classified error-string checks) above the phase-based catch-all:

```python
# Check stable docling status enums first (as today)
if docling_status == DoclingPhaseStatus.FAILED: ...
if docling_status == DoclingPhaseStatus.EXPIRED: ...

# Check globally-applicable error patterns before any phase catch-alls
if "already exists" in error:
    return {
        "component": "openrag",
        "failure_phase": "file_validation",
        "user_facing_message": "A file with this name already exists.",
        "actionable_by": "USER_ACTIONABLE",
    }

# Phase-specific and phase-dependent checks last
if phase == IngestionPhase.DOCLING and "Docling conversion did not complete" in error: ...
if phase == IngestionPhase.DOCLING and docling_status == DoclingPhaseStatus.PROCESSING: ...
if phase == IngestionPhase.LANGFLOW: ...

return None
```

File: src/services/docling_service.py [hunk 6]
```diff
@@ -33,6 +33,10 @@ class DoclingServeError(Exception):
     """Raised when docling-serve conversion fails."""
 
 
+class DoclingTransientError(DoclingServeError):
```
Comment (mpawlow) [inline src/services/docling_service.py:36]: ### (1d) [Normal] Two separate `DoclingServeError` class definitions — `DoclingTransientError` hierarchy only covers one

#### Problem

- `DoclingServeError` is independently defined in two modules:
  - `src/utils/docling_client.py:13` — legacy definition
  - `src/services/docling_service.py:32` — new definition
- `DoclingTransientError` subclasses **only** `services.docling_service.DoclingServeError`.
- Any caller that imports `DoclingServeError` from `utils.docling_client` and catches it will silently treat `DoclingTransientError` as a permanent error (since the two `DoclingServeError` classes are unrelated Python objects).
- The two classes have the same name, inviting future developers to mix up imports — especially as more code is added around this flow.

#### Background Information

- The duplicate definition appears pre-existing; this PR did not introduce it. However, the PR introduces a new type hierarchy (`DoclingTransientError`) on top of one of the two definitions, which increases the risk of cross-module confusion.

#### Code References

- `src/utils/docling_client.py:13`

#### Potential Solution

Consolidate: re-export `DoclingServeError` from one canonical location (e.g., `services/docling_service.py`) and have `utils/docling_client.py` import from there. This is a follow-up task given the pre-existing state, but the current PR should at minimum add a `# noqa` or a comment calling out the dual definition so it does not create silent surprises.

File: src/api/v1/documents.py [hunk 7]
```diff
@@ -64,6 +65,22 @@ async def ingest_endpoint(
     )
 
 
+async def all_tasks_enhanced_endpoint(
```
Comment (mpawlow) [inline src/api/v1/documents.py:66]: ### (1e) [Minor] `get_task_status2`/`get_all_tasks2` and `get_all_tasks2` filter completed files differently than `get_task_status2`

#### Problem

- `get_task_status2` (line 728) iterates **all** files in `upload_task.file_tasks` including completed ones — consistent with `get_task_status`.
- `get_all_tasks2` (line 777) explicitly skips completed files (`if file_task.status.value != "completed":`) — consistent with `get_all_tasks`.
- Consumers of the `/tasks/{task_id}/enhanced` endpoint (which calls `get_task_status2`) will see completed files in the `"files"` dict; consumers of `/tasks/enhanced` (which calls `get_all_tasks2`) will **not** see completed files.
- This behavioral asymmetry is present in the pre-existing methods and is perpetuated in the new "enhanced" variants without being documented in the endpoint docstrings, which could confuse API consumers.

#### Code References

- `src/services/task_service.py:728` — `get_task_status2`: all files included
- `src/services/task_service.py:777` — `get_all_tasks2`: completed files excluded
- `src/api/v1/documents.py:68–81` — `all_tasks_enhanced_endpoint` docstring (no mention of completed-file filtering)
- `src/api/v1/documents.py:96–114` — `task_status_enhanced_endpoint` docstring (no mention)

#### Potential Solution

Document the divergent behaviour in both endpoint docstrings:

```python
async def all_tasks_enhanced_endpoint(...):
    """...
    Note: completed files are omitted from each task's ``files`` dict to
    reduce payload size; use GET /v1/tasks/{task_id}/enhanced for the full
    file list of a specific task.
    """
```

**Review by mpawlow — APPROVED** [review]
@ricofurtado 

## Code Review 1

- ✅  Approved / LGTM  🚀 
- See: PR comments: 
   - (1a) needs to be addressed (Ruff failures)
   - (1b) to (1e) for optional consideration
-

File: src/services/docling_service.py (hunk 6 above)
Comment (ricofurtado) [reply src/services/docling_service.py:36]: It was pre-existing
